prime-router/docs/design/design/outer-resource-design.md

```diff
@@ -0,0 +1,78 @@
+```
+# resource: <Path to organization>
+# organization: <Path to organization>
+- name: performing-organization-name-pracrole
+  condition: '%organization.performer.resolve() is PractitionerRole'
+  resource: '%resource.performer.resolve().organization.resolve()'
+  schema: ./datatype/xon-organization
+  constants:
+    hl7OrgField: '%{hl7OBXField}-23'
+```
+
+With this new variable, the condition can work more as expected. But in some cases, it's not possible to reliably set that extra variable to point to the same resource as `resource`. The problem with this kind of workaround is illustrated by this code from `order-observation.yml`:
+```
+- name: observation-result-with-aoe
+  # Grab only the AOE observations from ServiceRequest.supportingInfo NOT associated with a specimen
+  resource: '%resource.result.resolve() | %service.supportingInfo.resolve().where(specimen.exists().not())'
+  schema: observation-result
+  resourceIndex: resultIndex
+  constants:
+    hl7ObservationPath: '/PATIENT_RESULT/ORDER_OBSERVATION(%{orderIndex})/OBSERVATION(%{resultIndex})'
+    hl7OBXField: '/PATIENT_RESULT/ORDER_OBSERVATION(%{orderIndex})/OBSERVATION(%{resultIndex})/OBX'
+    observation: '%diagnostic.result[%resultIndex].resolve()'
+```
+We want to be able to access both the initial observation (`'%diagnostic.result[%resultIndex].resolve()'`) as well as 
+the resource that is the result of this condition 
+`%resource.result.resolve() | %service.supportingInfo.resolve().where(specimen.exists().not())`. 
+However, to do this, we are needing to set the observation constant to the value of the 
+outer resource. This is a problematic workaround because this piece of code isn't actually working how you might 
+expect it to. Currently, if there are 5 items in`%resource.result.resolve()` and 2 items in 
+`%service.supportingInfo.resolve().where(specimen.exists().not())`, the 
+`resourceIndex` will go up to 7. The `observation` expression will be invalid when `resultIndex` is 6 or 7 since
+there are only 5 items in `%diagnostic.result` and that is ultimately what resource will end up being set to.
+
+We want to have a way to access that outer resource, the one initially defined, so that in places like these we can
+reference the parent a
+nd some of its children without losing the value of the initial resource.
+
+There are two ways we can go about this, and they are not mutually exclusive. 
+
+## Proposals
+### Outer Resource
+This is, by far, the less difficult option. In `FhirToHl7Converter`, we would set a constant called `outerResource` to 
```

Comment:
Smallest of nits, but curious what your thoughts on calling this `schemaResource` would be?  That would allow it to work for the root schema too (where schemaResource = Bundle)

Reply:
Either way works for me, thoughts @jimmyfagan?